### ISSUE TITLE
Validation loss exploding, potentially due to setting is_training = False

### ISSUE THREAD

**Issue by suryatejadev** [taehoonlee/tensornets#47]
Hi! Thanks a lot for this repo, I think it's very well written and extremely easy to use!

I am training a simple MNIST multi-class classifier using pretrained resnet50. I am using a placeholder 'is_train' to set the 'is_training' argument in the resnet50 model to True (during training) and False (during validation). However, when I set it's value to False during validation, my validation loss is exploding. This is not happening if I set it to True during validation. 

Please find the code to replicate this issue below. 
```
import numpy as np
import tensorflow as tf
import tensornets as tfnets
from tensorflow.examples.tutorials.mnist import input_data
np.random.seed(0)

graph = tf.Graph()
with graph.as_default():
    mnist = input_data.read_data_sets("mnist_data/", one_hot=False)

    x = tf.placeholder(tf.float32, [None, 256, 256, 3])
    y = tf.placeholder(tf.int32, [None])
    is_train = tf.placeholder(tf.bool)
    optimizer = tf.train.AdamOptimizer()

    model = tfnets.ResNet50(x, is_training = is_train, classes = 10)
    logits = model.get_outputs()[-2]
    probs = tf.nn.softmax(logits)
    loss = tf.reduce_mean(tf.nn.sparse_softmax_cross_entropy_with_logits(labels = y, logits = logits))
    grads = optimizer.compute_gradients(loss)
    train_op = optimizer.apply_gradients(grads)

    config = tf.ConfigProto()
    config.gpu_options.allow_growth = True
    config.allow_soft_placement = True
    with tf.Session(config=config) as sess:
        sess.run(tf.global_variables_initializer())
        sess.run(model.pretrained())

        print('Training ...')
        batch_size = 32
        for iteration in range(20):
            _x, _y = mnist.train.next_batch(batch_size)
            _x = np.resize(np.stack((np.reshape(_x, (-1, 28, 28)),)*3, axis = -1), (batch_size, 256, 256, 3))
            _, lossval = sess.run([train_op, loss], feed_dict = {x:_x, y:_y, is_train:True})
            print(iteration, lossval)

        print('Validation ...')
        for iteration in range(5):
            _x, _y = mnist.validation.next_batch(batch_size)
            _x = np.resize(np.stack((np.reshape(_x, (-1, 28, 28)),)*3, axis = -1), (batch_size, 256, 256, 3))
            lossval = sess.run(loss, feed_dict = {x:_x, y:_y, is_train:False})
            print(iteration, lossval)
```

The output I am getting when I set 'is_train' to False while calculating validation loss is as follows: 
```
Training ...
0 2.5664392
1 2.5185232
2 2.6779196
3 2.904881
4 2.5281467
5 2.3716216
6 2.6480408
7 2.4264488
8 2.5613265
9 2.6664178
10 2.407669
11 2.588695
12 2.4091241
13 2.6468048
14 2.6137376
15 2.3654165
16 2.3985069
17 2.302301
18 2.3154862
19 2.643575
Validation ...
0 3006.0938
1 3228.6968
2 3172.6226
3 2331.4446
4 2426.8574
```

However, when I set  'is_train' to True while calculating validation loss, the output is as follows:
```
Training ...
0 2.6028776
1 2.4225206
2 2.7544856
3 2.638253
4 2.4008565
5 2.642138
6 2.3750484
7 2.840887
8 2.3587594
9 2.6521862
10 2.450266
11 2.3560605
12 2.3225245
13 2.4954278
14 2.3524
15 2.3139462
16 2.2738576
17 2.4993274
18 2.3943353
19 2.7758856
Validation ...
0 2.2924833
1 2.396636
2 2.4049609
3 2.3891609
4 2.3445334
```

It would be great if you could let me know what's going wrong here. I am using tensorflow 1.13 and python 3.6. Thanks!

**Comment by suryatejadev** [comment]
I found my error. I was not adding batch norm variables to the optimizer, didn't realize I had to do that manually. So basically I changed ```train_op``` to : 
```
bn_update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
train_op = tf.group(optimizer.apply_gradients(grads), bn_update_ops)
```
That still lead to high validation loss values in the first 10-15 epochs, but then it started reducing until I reached >90% validation accuracy.
